service-automation/src/com/appyschool/suiteSetup/TestSuiteSetup.java

```diff
@@ -426,23 +428,87 @@ public void testCreateStudents(HashMap<String, String> map) throws JSONException
 		Response response = StudentServices.createStudent(json.toString());
 		response.then().log().all();
 		
 		JSONObject responseObj = new JSONObject(response.getBody().asString());
-		verifyEquals(response.statusCode(), 200);
-		verifyTrue(responseObj.get("id") != null);
-		verifyEquals(responseObj.getString("status"), "OK");
+		SoftAssertionBase.verifyEquals(response.statusCode(), 200);
+		SoftAssertionBase.verifyTrue(responseObj.get("id") != null);
+		SoftAssertionBase.verifyEquals(responseObj.getString("status"), "OK");
 		SuiteSetupProperties.setValue("STUDENT_ID_" + n, responseObj.get("id"));
 	}
 	
+	/**
+	 * Create Designation
+	 * 
+	 * @param map
+	 * @throws JSONException
+	 */
+
+		
+	@Test(dataProviderClass = com.appyschool.common.CommonUtils.class, dataProvider = "getDataFromFile")
+	@DataProviderArguments("./testData/configuration/createDesignation.csv")
+	public void testCreateDesignation(HashMap<String, String> map) throws JSONException {
+		l++;
+		String jsonTemplate = CommonUtils.getJsonTemplate("./testData/configuration/branches/json/createDesignation.json");
+		JSONObject json = CommonUtils.getJsonFromTemplate(jsonTemplate, map);
+		json.put("name",RandomStringUtils.randomAlphabetic(10));
+		json.put("description",RandomStringUtils.randomAlphabetic(10));
+		Response response = StaffServices.createDesignation(json.toString());
+		response.then().log().all();
+		System.out.print(response);
+		JSONObject responseObj = new JSONObject(response.getBody().asString());
+		SoftAssertionBase.verifyEquals(response.statusCode(), Integer.parseInt(map.get("statusCode")));
+		SoftAssertionBase.verifyTrue(responseObj.get("id") != null);
+		SuiteSetupProperties.setValue("Desig_ID_" + l, responseObj.get("id"));
+//		SoftAssertionBase.verifyTrue(responseObj.get("id") != null);
+	
+	}
+	
+	/**
+	 * Create a Staff
+	 * @param map
+	 * @throws JSONException
+	 */
+	@Test(dataProviderClass = com.appyschool.common.CommonUtils.class, dataProvider = "getDataFromFile" , dependsOnMethods="testCreateDesignation")
+	@DataProviderArguments("./testData/csv/staff1.csv")
+	public void testCreateStaff(HashMap<String, String> map)
+			throws JSONException {
+		String jsonTemplate = CommonUtils
+				.getJsonTemplate("./testData/json/straff1.json");
+		JSONObject json = CommonUtils.getJsonFromTemplate(jsonTemplate, map);
+		json.put("primaryPhoneNumber", RandomStringUtils.randomNumeric(10));
+		json.put("primaryEmailId",
+				map.get("$firstName") + RandomStringUtils.randomNumeric(2)
+						+ "@gmail.com");
+	json.put("staffdesignation",
+			map.get("$staffdesignation"));
+		json.put("idnumber",              
+				map.get("$idnumber"));    
+		json.put("firstname",              
+				map.get("$firstname"));             
+		json.put("gender",              
+				"M");      
+		json.put("addressLine1",              
+				RandomStringUtils.randomAlphabetic(10)); 
```

Comment:
can be read from the csv file. what is the need of adding it here?

Reply:
Fixed.